### PR TITLE
Gen3release

### PR DESCRIPTION
**Pull request by haraprasadj**
### Improvements
Added support for from_date and to_date

### CODE REVIEW

File: gen3git.py [hunk 1]
```diff
@@ -309,6 +320,15 @@ def main(args=None):
     # add 5 seconds to the stop date because the PR's "merged_at" date may
     # be a few seconds after the merged commit is created in master:
     stop_date = stop_commit.commit.author.date + timedelta(0, 5)
+
+    # Modifying for gen3release support to generate release notes based on the dates
+    from_date = getattr(args, "from_date", None)
+    if from_date:
```
Comment (paulineribeyre) [inline gen3git.py]: you can just do:
```
if args.from_date:
    start_date = datetime.strptime(args.from_date, "%Y-%m-%d")
```

File: gen3git.py [hunk 2]
```diff
@@ -309,6 +320,15 @@ def main(args=None):
     # add 5 seconds to the stop date because the PR's "merged_at" date may
     # be a few seconds after the merged commit is created in master:
     stop_date = stop_commit.commit.author.date + timedelta(0, 5)
+
+    # Modifying for gen3release support to generate release notes based on the dates
```
Comment (paulineribeyre) [inline gen3git.py]:
```suggestion
    # If dates are specified by the user, they override dates from tags/commits
```

File: gen3git.py [hunk 3]
```diff
@@ -167,6 +167,17 @@ def get_command_line_args():
         help="Tag to stop collecting release notes at (inclusive), default is $TRAVIS_TAG if set, "
         "or current git HEAD.",
     )
+    parser.add_argument(
+        "--from-date",
+        type=str,
+        help="Date to start getting release notes from (inclusive), format - YYYY-MM-DD.",
```
Comment (paulineribeyre) [inline gen3git.py]:
```suggestion
        help="Date to start getting release notes from (inclusive), format - YYYY-MM-DD. Overrides --from-tag argument. If not specified, falls back to default --from-tag.",
```

File: gen3git.py [hunk 4]
```diff
@@ -167,6 +167,17 @@ def get_command_line_args():
...
+    )
+    gen.add_argument(
+        "--to-date",
+        type=str,
+        help="Date to stop collecting release notes at (inclusive), format - YYYY-MM-DD, "
+             "default is $TRAVIS_TAG if set, or current git HEAD.",
```
Comment (paulineribeyre) [inline gen3git.py]:
```suggestion
             "Overrides --to-tag argument. If not specified, falls back to default --to-tag.",
```

File: gen3git.py (hunk 1 above)
Comment (haraprasadj) [reply gen3git.py]: Thanks, will do that

**Pull request merged**